src/backend/distributed/transaction/relation_access_tracking.c

```diff
@@ -173,7 +173,7 @@ RecordRelationAccessIfReferenceTable(Oid relationId, ShardPlacementAccessType ac
 	 * recursively calling RecordRelationAccessBase(), so becareful about
 	 * removing this check.
 	 */
-	if (PartitionMethod(relationId) != DISTRIBUTE_BY_NONE)
+	if (!IsCitusTableType(relationId, REFERENCE_TABLE))
```

Comment:
I'm not sure about this, though we can finalize this part with @onurctirtir tomorrow on this PR, let's merge this as-is for now